src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -332,6 +345,180 @@ def list_groups(client, display_name=None, query_filter=None):
+                                         available_to_other_tenant=False,
+                                         password=secret,
+                                         start_date=start_date,
+                                         end_date=end_date)
+    #pylint: disable=no-member
+    aad_sp = create_service_principal(client.service_principals, aad_application.app_id)
+    _build_output_content(name, aad_sp.object_id, secret, client.config.tenant_id)
+
+def reset_service_principal_credential(name, secret=None, years=1):
+    '''reset credential, on expiration or you forget it.
+
+    :param str name: the uri representing the name of the service principal
+    :param str secret: the secret used to login. If missing, command will generate one.
+    :param str years: Years the secret will be valid.
+    '''
+    client = _graph_client_factory()
+
+    #pylint: disable=no-member
+
+    #look for the existing application
+    query_exp = 'identifierUris/any(x:x eq \'{}\')'.format(name)
+    aad_apps = list(client.applications.list(filter=query_exp))
+    if not aad_apps:
+        raise CLIError('can\'t find an application matching \'{}\''.format(name))
+    #no need to check 2+ matches, as app id uri is unique
+    app = aad_apps[0]
+
+    #look for the existing service principal
+    query_exp = 'servicePrincipalNames/any(x:x eq \'{}\')'.format(name)
+    aad_sps = list(client.service_principals.list(filter=query_exp))
+    if not aad_sps:
+        raise CLIError('can\'t find a service principal matching \'{}\''.format(name))
+    sp_object_id = aad_sps[0].object_id
+
+    #build a new password credential and patch it
+    secret = secret or str(uuid.uuid4())
+    start_date = datetime.datetime.now()
+    end_date = start_date + relativedelta(years=years)
+    key_id = str(uuid.uuid4())
+    app_cred = PasswordCredential(start_date, end_date, key_id, secret)
+    app_create_param = ApplicationUpdateParameters(password_credentials=[app_cred])
+
+    client.applications.patch(app.object_id, app_create_param)
+
+    _build_output_content(name, sp_object_id, secret, client.config.tenant_id)
+
+def _build_output_content(sp_name, sp_object_id, secret, tenant):
+    logger.warning("Service principal has been configured with name: '%s', secret: '%s'",
+                   sp_name, secret)
+    logger.warning('Useful commands to manage azure:')
+    logger.warning('Assign a role:')
+    logger.warning('    az role assignment create --assignee %s --role Contributor', sp_object_id)
+    logger.warning('Log in:')
+    logger.warning('    az login --service-principal -u %s -p %s --tenant %s',
+                   sp_name, secret, tenant)
+    logger.warning('Reset credentials:')
+    logger.warning('    az ad sp reset-sp-credentials --name %s', sp_name)
+    logger.warning('Revoke:')
+    logger.warning('    az ad sp delete --id %s', sp_name)
+    logger.warning('    az ad app delete --id %s', sp_name)
```

Comment:
We thought about the output. The sample command snippet needs the result from executing the command, so people can copy, paste and run. Because of this, we just output it.